server.go

```diff
@@ -1488,6 +1491,156 @@ func (s *Server) ListenAndServe(addr string) error {
+// SetReadDeadline and SetWriteDeadline.
+//
+// A deadline is an absolute time after which I/O operations
+// fail instead of blocking. The deadline applies to all future
+// and pending I/O, not just the immediately following call to
+// Read or Write. After a deadline has been exceeded, the
+// connection can be refreshed by setting a deadline in the future.
+//
+// If the deadline is exceeded a call to Read or Write or to other
+// I/O methods will return an error that wraps os.ErrDeadlineExceeded.
+// This can be tested using errors.Is(err, os.ErrDeadlineExceeded).
+// The error's Timeout method will return true, but note that there
+// are other possible errors for which the Timeout method will
+// return true even if the deadline has not been exceeded.
+//
+// An idle timeout can be implemented by repeatedly extending
+// the deadline after successful Read or Write calls.
+//
+// A zero value for t means I/O operations will not time out.
+func (gc *GnetConn) SetDeadline(t time.Time) error {
+	return nil
+}
+
+// SetReadDeadline sets the deadline for future Read calls
+// and any currently-blocked Read call.
+// A zero value for t means Read will not time out.
+func (gc *GnetConn) SetReadDeadline(t time.Time) error {
+	return nil
+}
+
+// SetWriteDeadline sets the deadline for future Write calls
+// and any currently-blocked Write call.
+// Even if write times out, it may return n > 0, indicating that
+// some of the data was successfully written.
+// A zero value for t means Write will not time out.
+func (*GnetConn) SetWriteDeadline(t time.Time) error {
+	return nil
+}
+
+// GnetHTTP -
+type GnetHTTP struct {
+	*gnet.EventServer
+	fasthttpserver *Server
+}
+
+// OnOpened fires when a new connection has been opened.
+// The parameter:c has information about the connection such as it's local and remote address.
+// Parameter:out is the return value which is going to be sent back to the client.
+func (es *GnetHTTP) OnOpened(c gnet.Conn) (out []byte, action gnet.Action) {
+	log.Println("OPENED")
+	return
+}
+
+// OnClosed fires when a connection has been closed.
+// The parameter:err is the last known connection error.
+func (es *GnetHTTP) OnClosed(c gnet.Conn, err error) (action gnet.Action) {
+	log.Println("CLOSED")
+	return
+}
+
+// PreWrite fires just before any data is written to any client socket, this event function is usually used to
+// put some code of logging/counting/reporting or any prepositive operations before writing data to client.
+func (es *GnetHTTP) PreWrite() {
+	log.Println("PREWRITE")
+}
+
+// React fires when a connection sends the server data.
+// Call c.Read() or c.ReadN(n) within the parameter:c to read incoming data from client.
+// Parameter:out is the return value which is going to be sent back to the client.
+func (es *GnetHTTP) React(frame []byte, c gnet.Conn) (out []byte, action gnet.Action) {
+	log.Println("REACTING!!!!!")
+	gconn := &GnetConn{
+		gnetConn:    c,
+		readBuffer:  bytes.NewBuffer(frame),
```

Comment:
This is not going to work if the http request is spread over multiple packets in which case `React` might be called with a partial request in `frame` and `serveConn` will error.